server/src/main/java/com/ncc/neon/controllers/BetterController.java

```diff
@@ -1,70 +1,59 @@
 package com.ncc.neon.controllers;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ncc.neon.common.LanguageCode;
+import com.ncc.neon.common.NlpClientQueryBuilder;
+import com.ncc.neon.common.RemoteNlpClient;
 import com.ncc.neon.models.BetterFile;
 import com.ncc.neon.models.DataNotification;
+import com.ncc.neon.models.FileStatus;
+import com.ncc.neon.services.BetterFileService;
 import com.ncc.neon.services.DatasetService;
+import com.ncc.neon.services.FileShareService;
 import lombok.extern.slf4j.Slf4j;
-import okhttp3.*;
-import org.apache.http.HttpHost;
-import org.elasticsearch.action.admin.indices.refresh.RefreshRequest;
-import org.elasticsearch.action.admin.indices.refresh.RefreshResponse;
-import org.elasticsearch.action.delete.DeleteRequest;
-import org.elasticsearch.action.delete.DeleteResponse;
-import org.elasticsearch.action.get.GetRequest;
-import org.elasticsearch.action.get.GetResponse;
-import org.elasticsearch.client.RestClient;
 import org.elasticsearch.rest.RestStatus;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.core.env.Environment;
 import org.springframework.core.io.Resource;
 import org.springframework.core.io.UrlResource;
 import org.springframework.http.HttpHeaders;
-import okhttp3.Request;
-import okhttp3.Response;
-import org.elasticsearch.action.index.IndexRequest;
-import org.elasticsearch.action.index.IndexResponse;
-import org.elasticsearch.client.*;
 import org.springframework.http.HttpStatus;
-import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.http.codec.multipart.FilePart;
 import org.springframework.web.bind.annotation.*;
-import org.springframework.web.bind.annotation.ResponseBody;
 import org.springframework.web.reactive.function.client.WebClient;
-import reactor.core.publisher.Flux;
+import org.springframework.web.server.ResponseStatusException;
 import reactor.core.publisher.Mono;
-import reactor.util.function.Tuple2;
-import reactor.util.function.Tuples;
 
-import java.io.File;
-import java.io.IOException;
 import java.net.MalformedURLException;
 import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.util.*;
 
 @CrossOrigin(origins = "*")
 @RestController
 @RequestMapping("better")
 @Slf4j
 public class BetterController {
-    OkHttpClient client = new OkHttpClient();
-    ObjectMapper objectMapper = new ObjectMapper();
-    RestHighLevelClient elasticSearchClient;
     WebClient enPreprocessorClient;
     WebClient arPreprocessorClient;
     WebClient bpeClient;
-
+    WebClient nmtClient;
+    RemoteNlpClient enPreprocessorRNC;
+    RemoteNlpClient arPreprocessorRNC;
+    RemoteNlpClient bpeRNC;
+    RemoteNlpClient nmtRNC;
```

Comment:
I originally passed in the necessary Webclient to the `RemoteNlpClient.performNlpOperation` function, which would only require 1 RemoteNlpClient object. Upon discussing with @chase-farmer, he preferred this solution. I agree that this doesn't scale well as we incorporate more modules and prefer a cleaner solution. What do you mean by "something data-driven"?

Reply:
I meant where the modules are defined in a config file or a database so you could define new modules just with "data" rather than with code changes.

Reply:
I like this idea.  It would streamline the process of "registering" a new NLP module.  My question then is how would these modules be instantiated/referenced in the code?  Perhaps a static getter that takes some identifier?

Reply:
I think a singleton rather than a static method, but I guess something like that.

Reply:
Created new ticket BET-50 to handle this task.